File: public/docs/_examples/security/ts/app/bypass-security.component.ts
// #docplaster
// #docregion
import {Component, OnInit} from '@angular/core';
import {DomSanitizationService, SafeResourceUrl, SafeUrl} from '@angular/platform-browser';

@Component({
  selector: 'bypass-security',
  templateUrl: 'app/bypass-security.component.html',
})
export class BypassSecurityComponent implements OnInit {
  dangerousUrl: SafeUrl;
  videoId: string = 'PUBnlbjZFAI';  // bound to the videoId input box.

  // #docregion trust-url
  constructor(private sanitizer: DomSanitizationService) {
    // javascript: URLs are dangerous if attacker controlled. Angular sanitizes them in data
    // binding, but we can explicitly tell Angular to trust this value:
    this.dangerousUrl = sanitizer.bypassSecurityTrustUrl('javascript:alert("Hi there")');
  }
  // #enddocregion trust-url

  // #docregion trust-video-url
  trustedVideoUrl(id: string): SafeResourceUrl {
    // Appending an ID to a YouTube URL is safe.
    // Always make sure to construct SafeValue objects as close as possible to the input data, so
    // that it's easier to check if the value is safe.
    return this.sanitizer.bypassSecurityTrustResourceUrl('https://www.youtube.com/embed/' + id);
  }
  // #enddocregion trust-video-url
}
// #enddocregion

File: public/docs/_examples/security/dart/lib/bypass_security_component.dart
// #docplaster
// #docregion
import 'package:angular2/core.dart';

/// NOTE: this implementation is incomplete. Current content is provided
/// as a placeholder.
@Component(
    selector: 'bypass-security',
    templateUrl: 'bypass_security_component.html')
class BypassSecurityComponent {
  // DomSanitizationService sanitizer;
  String dangerousUrl;
  dynamic /*SafeUrl*/ trustedUrl;
  String dangerousVideoUrl;
  dynamic /*SafeResourceUrl*/ videoUrl;

  // #docregion trust-url
  BypassSecurityComponent(/*this.sanitizer*/) {
    // javascript: URLs are dangerous if attacker controlled.
    // Angular sanitizes them in data binding, but we can
    // explicitly tell Angular to trust this value:
    dangerousUrl = 'javascript:alert("Hi there")';
    trustedUrl = /*sanitizer.bypassSecurityTrustUrl*/('javascript:alert("Hi there")');
    // #enddocregion trust-url
    updateVideoUrl('PUBnlbjZFAI');
  }
  // #docregion trust-video-url
  void updateVideoUrl(String id) {
    // Appending an ID to a YouTube URL is safe.
    // Always make sure to construct SafeValue objects as
    // close as possible to the input data, so
    // that it's easier to check if the value is safe.
    dangerousVideoUrl = 'https://www.youtube.com/embed/$id';
    videoUrl = /*sanitizer.bypassSecurityTrustResourceUrl*/(dangerousVideoUrl);
  }
  // #enddocregion trust-video-url
}
